claat/render/md.go

```diff
@@ -243,24 +243,22 @@ func (mw *mdWriter) itemsList(n *types.ItemsListNode) {
 }
 
 func (mw *mdWriter) infobox(n *types.InfoboxNode) {
-	// TODO: This should use the "detail item" syntax so that it can be pure MD and not HTML
-	// kind
-	// : <content>
-	//
-	// The main issue is that when you do write(n.Content.Nodes...) it always adds two newlines
-	// at the beginning. This is because the gdoc parser parses info boxes as a list node
-	// followed by a text node, and the list node is an md block unto itself, resulting in two
-	// newlines being added to the output before the text.
+	// InfoBoxes are comprised of a ListNode with the contents of the InfoBox. 
+	// Writing the ListNode directly results in extra newlines in the md output
+	// which breaks the formatting. So instead, write the ListNode's children 
+	// directly and don't write the ListNode itself.
+	ln, ok := n.Content.Nodes[0].(*types.ListNode)
+	if !ok {
+		return
```

Comment:
I feel like I need to do something better here, but we don't have a pattern for write methods raising errors. Thoughts?

Reply:
You could add something like docState (see the MD parser) that would report the errors at the end, but given how large an undertaking that would be, I don't think it's necessarily required here. Please do file an issue about it though, we should have better error handling overall.

Reply:
Done - https://github.com/googlecodelabs/tools/issues/452